pkg/porter/parameters.go

```diff
@@ -307,6 +307,23 @@ func (p *Porter) loadParameterSets(params []string) (valuesource.Set, error) {
 			return nil, err
 		}
 
+		// A parameter may correspond to a Porter-specific parameter type of 'file'
+		// If so, add value (filepath) directly to map and remove from pset
+		for _, paramDef := range p.Manifest.Parameters {
+			if paramDef.Type == "file" {
+				for i, param := range pset.Parameters {
+					if param.Name == paramDef.Name {
+						// Pass through value (filepath) directly to resolvedParameters
+						resolvedParameters[param.Name] = param.Source.Value
+						// Eliminate this param from pset to prevent its resolution by
```

Comment:
What if instead of loading the manifest here and trying to check the parameters before it gets to the cnab-go code, we wrapped the cnab-go host secret store, and extended it in porter to support our custom file type property?

Do you think it would be possible to change our support for the file type from a "hack" (that we shim in wherever we can fit it) to instead something that we override and support in the same struct/place that cnab-go supports the other types?

Reply:
It may be possible, but my initial concern on going this direction was that this `file` type has classically been exclusive to Porter.  As `cnab-go` is intended to be the Golang reference library for the CNAB Spec, which doesn't mention this type, I didn't think it was a proper fit.  However, I agree: it would be much cleaner to move support/override-ability into this library.

Say we do intend to proceed with adding to the cnab-go library.  As there is pre-existing code to clean up (not intro'd in this PR), I'd definitely like to work off of a new issue mentioning this broader scope.  The question would then be: do we want this fix (or a variant thereof) in the near-term to ensure functionality until the refactor (and corresponding cnab-go PR(s)) are scheduled to be worked on?

Reply:
Oops, miscommunication here. I was suggesting that we wrap the type defined in cnab-go responsible for resolving the parameter set and then extend it "inside of porter". So the file type would continue to be porter only.

Reply:
Ah, thank you.  I think I have a better understanding now.  I'll work with you to outline a strategy in a follow-up refactor in this area.